spec/consumer_resource_spec.rb

```diff
@@ -1553,6 +1551,82 @@ def verify_installed_pids(consumer, product_ids)
     }.to raise_exception(RestClient::BadRequest)
   end
 
+  it 'should create consumer with multiple environments' do
+    user_cp = user_client(@owner1, random_string('billy'))
+    env1 = @cp.create_environment(@owner1['key'], random_string("env_id-1"), random_string("env_name-1"))
+    env2 = @cp.create_environment(@owner1['key'], random_string("env_id-2"), random_string("env_name-2"))
+
+    consumer = user_cp.register(random_string('system'), :system, nil, {}, nil, nil, [], [], [{'id' => env1.id}, {'id' => env2.id}])
+    consumer_client = Candlepin.new(nil, nil, consumer['idCert']['cert'], consumer['idCert']['key'])
+    expect(consumer.environments.size).to eq(2)
+  end
+
+  it 'should not create consumer with nil environment in environments list' do
+    user_cp = user_client(@owner1, random_string('billy'))
+    env1 = @cp.create_environment(@owner1['key'], random_string("env_id-1"), random_string("env_name-1"))
+
+    lambda do
+      consumer = user_cp.register(random_string('system'), :system, nil, {}, nil, nil, [], [], [{'id' => env1.id}, {'id' => ''}])
+    end.should raise_exception(RestClient::ResourceNotFound)
+  end
+
+  it 'should update consumer with re-prioritized environments' do
+    user_cp = user_client(@owner1, random_string('billy'))
+    env1 = @cp.create_environment(@owner1['key'], random_string("env_id-1"), random_string("env_name-1"))
+    env2 = @cp.create_environment(@owner1['key'], random_string("env_id-2"), random_string("env_name-2"))
+
+    consumer = user_cp.register(random_string('system'), :system, nil, {}, nil, nil, [], [])
+    consumer_client = Candlepin.new(nil, nil, consumer['idCert']['cert'], consumer['idCert']['key'])
+
+    consumer_client.update_consumer({:environments => [{:name => env1.name}, {:name => env2.name}]})
+    consumer = @cp.get_consumer(consumer['uuid'])
+    expect(consumer.environments[0].name).to eq(env1.name)
+    expect(consumer.environments[1].name).to eq(env2.name)
+    # re-prioritized the environments
+    consumer_client.update_consumer({:environments => [{:name => env2.name}, {:name => env1.name}]})
+    consumer = @cp.get_consumer(consumer['uuid'])
+    expect(consumer.environments[0].name).to eq(env2.name)
+    expect(consumer.environments[1].name).to eq(env1.name)
+
+  end
+
+  it 'should update consumer with updated environments' do
```

Comment:
Let's make the name of this test more descriptive of what it tests. Something like 'should preserve other environment priorities on a consumer when an environment is removed'

Reply:
Done.